_posts/2020-07-21-cube-functionality-to-kartothek.markdown

```diff
@@ -0,0 +1,409 @@
+...     datasets=["transformed"],
+... )
+>>> query_cube(
+...     cube=cube,
+...     store=store,
+... )[0]
+   avg_temp      city country        day   latitude  longitude
+0         8   Dresden      DE 2020-07-01  51.050407  13.737262
+1         4   Dresden      DE 2020-07-02  51.050407  13.737262
+2         6   Hamburg      DE 2020-07-01  53.551086   9.993682
+3         5   Hamburg      DE 2020-07-02  53.551086   9.993682
+4         6    London      UK 2020-07-01        NaN        NaN
+5         8    London      UK 2020-07-02        NaN        NaN
+6        20  Santiago      CL 2020-07-01        NaN        NaN
+7        22  Santiago      CL 2020-07-02        NaN        NaN
+```
+
+## Cube Features in Kartothek
+
+*	**Multiple-datasets**: When mapping multiple parts (tables or datasets) to Kartothek, using multiple datasets allow users to copy, backup and delete them separately.
+ Index structures are bound to datasets. This was not possible with the existing multi-table (within a single dataset) feature present in kartothek.
+ We intend to phase out the multi-table single dataset functionality soon.
+
+*	**Seed-Based Join System / Partition-alignment**: When data is stored in multiple parts (tables or datasets), the question is how to expose it to the user during read operations.
+ Seed-based join marks a single part as seed which provides seed dataset in the cube, all other parts are just additional columns.
+ Cube uses lazy approach of seed based join, since it better supports independent copies and backups of datasets and also simplifies some of our processing pipelines
+ (e.g. geolocation data can blindly be fetched for too many locations and dates.)	
+
+
+## Outlook
+In the upcoming months we’ll continue to expand the Kartothek functionality. Here are a few highlights of what's next:
+
+- **API cleanup:** The API surface of kartothek grew organically over the years and we plan to re-design it. 
+While doing so, we will incorporate our learnings regarding API design and will also prune some features that are not needed anymore or that did not match their expectations (e.g. the original multi-table design).
+- **Ecosystem integration:** At this point in time, there are multiple dataset formats (e.g. [Apache Arrow](https://arrow.apache.org/docs/python/dataset.html), 
+[Apache Iceberg](https://iceberg.apache.org/), [Delta Lake](https://delta.io/)) and we will investigate how to evolve kartothek as a library and as a format to align better with the ecosystem and enable new features (like schema migrations and time travel), 
+while providing the stability and safety that our users require.
+- **Query Planning:** Currently the kartothek query planner solely relies on file-level information (file names for primary indices and separate index files for secondary indices). 
+It would be great to also use the RowGroup-level statistics as specified in [Apache Parquet](https://github.com/apache/parquet-format) to improve query performance.
```

Comment:
```suggestion
It would be great to also use the RowGroup-level statistics as specified by [Apache Parquet](https://github.com/apache/parquet-format) to improve query performance.
```